pkg/controller/rsm/transformer_status.go

```diff
@@ -52,91 +49,37 @@ func (t *ObjectStatusTransformer) Transform(ctx graph.TransformContext, dag *gra
 
 	switch {
 	case model.IsObjectUpdating(rsmOrig):
-		// use rsm's generation wo of sts's
+		// use rsm's generation instead of sts's
 		rsm.Status.ObservedGeneration = rsm.Generation
 	case model.IsObjectStatusUpdating(rsmOrig):
-		if rsm.Spec.RsmTransformPolicy == v1alpha1.ToPod {
-			ml := GetPodsLabels(rsm.Labels)
-			pods := &corev1.PodList{}
-			if err := transCtx.Client.List(transCtx, pods, client.InNamespace(rsm.Namespace), ml); err != nil {
-				return err
-			}
-			fliteredPods := FilterActivePods(pods.Items)
-			rsm.Status.Replicas = int32(len(fliteredPods))
-			readyReplicasCount, availableReplicasCount := calculateStatus(rsm, fliteredPods)
-			rsm.Status.ReadyReplicas = int32(readyReplicasCount)
-			rsm.Status.AvailableReplicas = int32(availableReplicasCount)
-			rsm.Status.UpdatedReplicas = rsm.Status.Replicas
-
-			// update role fields
-			setMembersStatus(rsm, pods.Items)
-		} else {
-			// read the underlying sts
-			sts := &apps.StatefulSet{}
-			if err := transCtx.Client.Get(transCtx.Context, client.ObjectKeyFromObject(rsm), sts); err != nil {
-				return err
-			}
-			// keep rsm's ObservedGeneration to avoid override by sts's ObservedGeneration
-			generation := rsm.Status.ObservedGeneration
-			rsm.Status.StatefulSetStatus = sts.Status
-			rsm.Status.ObservedGeneration = generation
-			if currentGenerationLabel, ok := sts.Labels[rsmGenerationLabelKey]; ok {
-				currentGeneration, err := strconv.ParseInt(currentGenerationLabel, 10, 64)
-				if err != nil {
-					return err
-				}
-				rsm.Status.CurrentGeneration = currentGeneration
-			}
-			// read all pods belong to the sts, hence belong to the rsm
-			pods, err := getPodsOfStatefulSet(transCtx.Context, transCtx.Client, sts)
+		// read the underlying sts
+		sts := &apps.StatefulSet{}
+		err := transCtx.Client.Get(transCtx.Context, client.ObjectKeyFromObject(rsm), sts)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+		// keep rsm's ObservedGeneration to avoid override by sts's ObservedGeneration
+		generation := rsm.Status.ObservedGeneration
+		rsm.Status.StatefulSetStatus = sts.Status
+		rsm.Status.ObservedGeneration = generation
+		if currentGenerationLabel, ok := sts.Labels[rsmGenerationLabelKey]; ok {
```

Comment:
If sts is null here? Above, only the isNotFound error is handled.

Reply:
nice catch, fixed in: [851aed9](https://github.com/apecloud/kubeblocks/pull/6776/commits/851aed915d2d9441ac101643776390a6417f5ad1)